engine/src/main/java/org/terasology/rendering/nui/widgets/ResettableUIText.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.widgets;
+
+import org.terasology.math.geom.Rect2i;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.SubRegion;
+import org.terasology.rendering.nui.UIWidget;
+
+public class ResettableUIText extends UIText {
+    private UIButton clearButton = new UIButton("clearButton", "X");
+
+    @Override
+    public void onDraw(Canvas canvas) {
+        //super.onDraw(canvas);
+        if (super.text.get() == null) {
+            super.text.set("");
+        }
+        super.lastFont = canvas.getCurrentStyle().getFont();
+        super.lastWidth = (int) (canvas.size().x * 0.8f);
+        if (isEnabled()) {
+            canvas.addInteractionRegion(super.interactionListener, Rect2i.createFromMinAndMax(0, 0, (int) (canvas.size().x * 0.8f), canvas.size().y));
+            clearButton.subscribe(new ActivateEventListener() {
```

Comment:
This should be lambdified:

```java
clearButton.subscribe(button -> setText(""));
```

IDEA should prompt you to convert anonymous inner classes to lambdas - do this whenever it does 🙂